PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/runner/webapp/webappconfig/RiderWebAppConfiguration.kt

```diff
@@ -0,0 +1,329 @@
+                .sortedWith(compareBy({ it.isWeb }, { it.projectName })).firstOrNull()
+    }
+
+    override fun getModel(): AzureDotNetWebAppSettingModel {
+        return myModel
+    }
+
+    override fun getConfigurationEditor(): SettingsEditor<out RunConfiguration> {
+        return RiderWebAppSettingEditor(project, this)
+    }
+
+    @Throws(ExecutionException::class)
+    override fun getState(executor: Executor, executionEnvironment: ExecutionEnvironment): RunProfileState? {
+        return RiderWebAppRunState(project, myModel)
+    }
+
+    override fun validate() { }
+
+    /**
+     * Validate the configuration to run
+     *
+     * @throws [RuntimeConfigurationError] when configuration miss expected fields
+     */
+    @Throws(RuntimeConfigurationError::class)
+    override fun checkConfiguration() {
+        validateSignIn()
+        validateProject()
+
+        if (myModel.isCreatingWebApp) {
+            validateWebAppName(myModel.webAppName)
+            checkValueSet(myModel.subscriptionId, SUBSCRIPTION_MISSING)
+
+            if (myModel.isCreatingResourceGroup) {
+                validateResourceGroupName(myModel.resourceGroupName)
+            } else {
+                checkValueSet(myModel.resourceGroupName, RESOURCE_GROUP_MISSING)
+            }
+
+            if (myModel.isCreatingAppServicePlan) {
+                checkValueSet(myModel.region, LOCATION_MISSING)
+                checkValueSet(myModel.pricingTier, PRICING_TIER_MISSING)
+                validateAppServicePlanName(myModel.appServicePlanName)
+            } else {
+                checkValueSet(myModel.appServicePlanId, APP_SERVICE_PLAN_MISSING)
+            }
+        } else {
+            checkValueSet(myModel.webAppId, WEB_APP_MISSING)
+        }
+    }
+
+    //region Sign In
+
+    /**
+     * Check whether user is signed in to Azure account
+     *
+     * @throws [ConfigurationException] in case validation is failed
+     */
+    @Throws(RuntimeConfigurationError::class)
+    private fun validateSignIn() {
+        try {
+            if (!AuthMethodManager.getInstance().isSignedIn) {
+                throw RuntimeConfigurationError(SIGN_IN_REQUIRED)
+            }
+        } catch (e: IOException) {
+            throw RuntimeConfigurationError(SIGN_IN_REQUIRED)
+        }
+    }
+
+    //endregion Sign In
+
+    //region Project
+
+    /**
+     * Validate publishable project selection in the config
+     * Check for the WebApplication targets that contains tasks for generating publishable package on Windows only
+     *
+     * @throws [ConfigurationException] in case validation is failed
+     */
+    @Throws(RuntimeConfigurationError::class)
+    private fun validateProject() {
+
+        val project = myModel.publishableProject ?: throw RuntimeConfigurationError(PROJECT_MISSING)
+
+        if (!project.isDotNetCore && !isWebTargetsPresent(File(project.projectFilePath)))
```

Comment:
add a comment that this is handling a specific case

Reply:
@IvanPashchenko, please see a comment about the method. It describes the validation. Updated a bit.